### PR TITLE
Add gitstatus segment

### PR DESCRIPTION
**Pull request by romkatv**
If [gitstatus plugin](https://github.com/romkatv/gitstatus) is enabled and the current directory is within a git repo, this segment displays the same prompt as vcs segment but 3 times faster. Otherwise it does nothing. Just like vcs, this segment is async. It talks to git in a worker process.

You can test it in Docker:

```zsh
docker run -e LANG=C.UTF-8 -e LC_ALL=C.UTF-8 -e TERM=$TERM -it --rm ubuntu bash -c '
  set -uex
  apt update
  apt install -y zsh git
  cd
  git clone https://github.com/romkatv/gitstatus.git
  git clone -b gitstatus https://github.com/romkatv/p10k
  echo "
    source ~/gitstatus/gitstatus.plugin.zsh
    source <(sed -e s/vcs/gitstatus/ ~/p10k/presets/robobenklein.p10kp.zsh)
    source ~/p10k/p10k.zsh" >~/.zshrc
  cd p10k
  zsh -i'
```

Hold `[ENTER]` and observe the fraction of prompts that have git info.

For comparison, do the same with the stock vcs segment.

```zsh
docker run -e LANG=C.UTF-8 -e LC_ALL=C.UTF-8 -e TERM=$TERM -it --rm ubuntu bash -c '
  set -uex
  apt update
  apt install -y zsh git
  cd
  git clone https://github.com/robobenklein/p10k
  echo "
    source ~/p10k/presets/robobenklein.p10kp.zsh
    source ~/p10k/p10k.zsh" >~/.zshrc
  cd p10k
  zsh -i'
```

On my machine I get about 3 times more prompts with git info per second when using gitstatus, which means it's producing data 3 times faster.

### CODE REVIEW

**Comment by robobenklein** [comment]
Awesome! You'll have to give me a bit of time to test this on the really old zsh versions, and I wonder how hard it would be to make this a detect-upgrade in the default vcs segment?

I think it would be great to have p10ks_vcs detect this when available, but I'd have to come up with some solution for subversion repos...

Anyways, I'm almost definitely going to rename P10K to something else and I'll probably try to get this merged in before I do the rename. (If you have name ideas - let me know, P10K doesn't really fit since there's no powerlevel9k code in here at all...)

**Comment by romkatv** [comment]
I sent you another PR -- an alternative to this one. See https://github.com/robobenklein/p10k/pull/2.

File: segments/p10ks_gitstatus_async [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  else
+    if [[ -n "$VCS_STATUS_REMOTE_BRANCH" &&
+          "$VCS_STATUS_LOCAL_BRANCH" != "$VCS_STATUS_REMOTE_BRANCH" ]]; then
+      echo -nE $' \u2192 '"$VCS_STATUS_REMOTE_BRANCH"
+    fi
+    local icons=''
+    (( VCS_STATUS_HAS_STAGED == 1 )) && icons+='✚'
+    (( VCS_STATUS_HAS_UNSTAGED == 1 )) && icons+='●'
+    (( VCS_STATUS_HAS_UNTRACKED == 1 )) && icons+='◌'
+    echo -nE "${icons:+ $icons}"
```
Comment (robobenklein) [inline segments/p10ks_gitstatus_async:20]: dang this is so much cleaner than using vcs_info...

**Review by robobenklein — APPROVED** [review]
I like dis

**Pull request merged**